### ISSUE TITLE
Add Update Button

### ISSUE THREAD

**Issue by 0xsingular** [teaxyz/teaBASE#20]
Description:
A "Check for Updates" button would be useful to let users know if a newer version of teaBASE is available. If an update is found, there could be an "Update Now" button to apply it easily.

Benefit:
Makes staying up to date simple and ensures users always have the latest features and fixes.

**Comment by stevenlei** [comment]
I have just check the code and it seems there is already a self update feature in place

So I installed an older version 0.5.0, and check the log with the system Console app, it will check for Github release and know 0.6.0 is the latest version:
<img width="1190" alt="image" src="https://github.com/user-attachments/assets/ced72f8f-4827-4d9d-91d8-e796d4947326" />

Although it is not reflected in real time, but after a few restarts it has upgraded successfully from 0.5.0 to 0.6.0

It could be silently failed sometimes, or just need time to reload the prefpane, not sure.

**Comment by mxcl** [comment]
We do indeed already self-update, there is no UI or notifications for this however and I agree there should be. Just haven’t gotten to it yet. Not sure how this should be displayed etc.

**Comment by mxcl** [comment]
Additionally, the update needs to be safe. Currently if System Settings Quits during update it will result in a partial update that is thus corrupted.

Atomically updating the prefPane is maybe possible. Currently we `rsync` over the top of the running prefPane. Moving the open prefPane may be ok, but I dunno if the system is going to like that.

**Comment by krisevan12** [comment]
```javascript
const { autoUpdater } = require('electron-updater');
const { ipcMain, BrowserWindow } = require('electron');

// Check for updates
ipcMain.handle('check-for-updates', () => {
  autoUpdater.checkForUpdates();
});

// Listen for update availability
autoUpdater.on('update-available', () => {
  // Show the "Update Now" button
  mainWindow.webContents.send('update-available');
});

// Apply update
autoUpdater.on('update-downloaded', () => {
  autoUpdater.quitAndInstall();
});
```

This code checks for updates and notifies the main window when an update is available. The app can be set to download and apply updates automatically.

**Comment by krisevan12** [comment]
should work, check

**Comment by all-landing** [comment]
> Additionally, the update needs to be safe. Currently if System Settings Quits during update it will result in a partial update that is thus corrupted.
> 
> Atomically updating the prefPane is maybe possible. Currently we `rsync` over the top of the running prefPane. Moving the open prefPane may be ok, but I dunno if the system is going to like that.

you are right - the update should be atomic, otherwise in case of a failure (e.g. if System Settings terminates during the update) you may get a corrupted or partially updated prefPane.

Possible approaches for a safe (atomic) update of the prefPane:

**Update via a temporary file and replace:**
Instead of immediately copying (rsync) over a running prefPane, you can first copy the new version to a separate temporary file (e.g., MyPane.prefPane.new). After a successful copy and integrity check - replace the original with an atomic operation (e.g. via mv or similar mechanism). This minimizes the window when the prefPane is in a non-consistent state. This approach is often used to update applications and configs on Unix systems.

**Lock/Use Check:**
Before replacing, it is worth making sure that prefPane is not in use (or at least alerting the user to close System Settings). If this is not possible, at least notify the user of the risks.

**Failure response:**
If the replacement was interrupted, the next time you run it, you can check for a temporary file and try to restore or roll up the update, or roll back to a previous version.

**Use of application flags:**
As Apple recommends, for complex customizations, you can save changes to a temporary document and apply them only after confirmation (via the Apply button). This reduces the risk of losing or corrupting data in the event of a crash

**Comment by mxcl** [comment]
lol fuck off ai bots
